src/Components/Toast.jsx

```diff
@@ -26,51 +27,71 @@ function Toast({ message, handleMessageRemove }) {
       clearTimeout(timeoutIdForRemoval);
       clearTimeout(timeoutIdForAnimation);
     };
-  }, []);
+  }, [handleMessageRemove, message.id]);
 
   const styles = {};
 
+  const handleCloseBtn = useCallback(() => {
+    setSlideOut(true);
+    setTimeout(() => {
+      handleMessageRemove(message.id);
+    }, 450);
+  }, [message.id, handleMessageRemove]);
+
   const toastIcon = () => {
```

Comment:
@Meet-Kalani Move this outside the component to avoid re-creation of this function on re-render